WalletWasabi/Logging/Logger.cs

```diff
@@ -29,24 +29,28 @@ public static class Logger
 
 	private static LogLevel MinimumLevel { get; set; } = LogLevel.Critical;
 
-	private static HashSet<LogMode> Modes { get; } = new HashSet<LogMode>();
+	private static HashSet<LogMode> Modes { get; } = new();
 
 	public static string FilePath { get; private set; } = "Log.txt";
 
-	public static string EntrySeparator { get; private set; } = Environment.NewLine;
+	public static string EntrySeparator { get; } = Environment.NewLine;
 
 	/// <summary>
 	/// Gets the GUID instance.
-	///
 	/// <para>You can use it to identify which software instance created a log entry. It gets created automatically, but you have to use it manually.</para>
 	/// </summary>
 	private static Guid InstanceGuid { get; } = Guid.NewGuid();
 
-	/// <summary>
-	/// Gets or sets the maximum log file size in KB.
-	/// </summary>
-	/// <remarks>Default value is approximately 10 MB. If set to <c>0</c>, then there is no maximum log file size.</remarks>
-	private static long MaximumLogFileSize { get; set; } = 10_000;
+	/// <summary>Gets or sets the maximum log file size in bytes.</summary>
+	/// <remarks>
+	/// Default value is approximately 10 MB. If set to <c>0</c>, then there is no maximum log file size.
+	/// <para>Guarded by <see cref="Lock"/>.</para>
+	/// </remarks>
+	private static long MaximumLogFileSizeBytes { get; set; } = 10_000_000;
```

Comment:
Why do we need to guard this property with a lock?
3 references: 1 write - 2 read.

`FilePath` property is a bit similar to this. 
It has  8 references: 1 write - 6 read -1 documentation and it is not guarded by a lock.